covasim/sim.py

```diff
@@ -417,18 +445,24 @@ def run(self, do_plot=False, verbose=None, **kwargs):
 
         T = sc.tic()
 
+        self.scaling_vec = [1.0]
+
         # Reset settings and results
         if not self.initialized:
             self.initialize()
 
         if verbose is None:
             verbose = self['verbose']
 
+        current_scale = 1.0
         # Main simulation loop
         for t in range(self.npts):
+            # Check if we need to rescale
+            if self["resample"]:
+                current_scale = self.resample(current_scale)
+            self.scaling_vec.append(current_scale)
```

Comment:
Would suggest just `self.resample()` which checks `if self["resample"]:` to begin with and then modifies `self.scaling_vec` in place